src/Parser/Core/Modules/Spells/BFA/AzeriteTraits/CrystallineCarapace.js

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import SpellIcon from 'common/SpellIcon';
+import { formatNumber, formatPercentage } from 'common/format';
+import StatisticBox, { STATISTIC_ORDER } from 'Interface/Others/StatisticBox';
+import Analyzer from 'Parser/Core/Analyzer';
+import { calculateAzeriteEffects } from 'common/stats';
+import SPELLS from 'common/SPELLS';
+
+export function crystallineCarapaceStats(combatant) {
+  if(!combatant.hasTrait(SPELLS.CRYSTALLINE_CARAPACE.id)) {
+    return null;
+  }
+  let armor = 0;
+  let thornDamage = 0;
+  for(const rank of combatant.traitsBySpellId[SPELLS.CRYSTALLINE_CARAPACE.id]) {
+    const [td, ar] = calculateAzeriteEffects(SPELLS.CRYSTALLINE_CARAPACE.id, rank);
+    armor += ar;
+    thornDamage += td;
+  }
+
+  return {armor, thornDamage};
+}
+
+export const STAT_TRACKER = {
+  armor: (combatant) => crystallineCarapaceStats(combatant).armor,
+  thornDamage: (combatant) => crystallineCarapaceStats(combatant).thornDamage,
+};
+
+class CrystallineCarapace extends Analyzer {
+  armor = 0;
+  thornDamage = 0;
+  constructor(...args) {
+    super(...args);
+    const resp = crystallineCarapaceStats(this.selectedCombatant);
+    console.log(resp);
+
+    if(resp === null) {
+      this.active = false;
```

Comment:
Please make it more obvious what the `active` check depends on to make the code easier to scan/follow.